mozilla/pdf.js
# Directly print PDFs in Chromium (+PDF editing functionality)

This PR adds a minimalistic and low-level `PDFDataWriter` class to PDF.js, which allows us to (incrementally) update PDFs. With this new class, I add print instructions to the PDF, which significantly improves the printing performance in Chromium-based browsers with a built-in PDF previewer (so, basically every Chromium browser AFAIK).

I suggest to separately review each commit, instead of checking the combined diff of the PR, because each commit is focused on a specific feature and has a detailed commit message.

To verify that the whole works as expected:
1. `node make chromium`
2. Open Chrome, visit chrome://extensions and load the unpacked extension from build/chromium/.
3. Open any PDF (preferably one with 10+ pages) with Chrome.
4. Focus the page and press Ctrl + P.
5. A print preview should immediately be displayed, without rendering all pages upfront.
6. Close the preview and click on the print button (or Ctr+P).
7. Same as 5.

and
1. `node make generic`
2. `node make server` (to start the server).
3. Visit localhost:8888 and open a PDF
4. See step 4 - 7 of the previous list.

and
1. Visit `http://localhost:8888/test/unit/unit_test.html?spec=pdf_data_writer` and verify that all tests are green.
